client/src/features/createPost/components/CreatePostModal.jsx

```diff
@@ -35,25 +78,49 @@ const CreatePostModal = ({ isOpen, setIsOpen, firstName, children }) => {
                 <h1 className="text-gray-900 dark:text-dark-txt font-semibold capitalize">
                   {firstName}
                 </h1>
-                <CreatePostSelect />
+                <CreatePostSelect select={select} setSelect={setSelect} />
               </div>
             </div>
-            <div className="form-control">
+            <div className="form-control px-3 max-h-80 scrollbar-thin hover:scrollbar-thumb-gray-300 dark:hover:scrollbar-thumb-dark-hover scrollbar-track-transparent overflow-y-auto">
               <textarea
                 name="text"
                 value={formData.text}
                 onChange={handleInputChange}
                 className="textarea h-24 textarea-ghost p-0 text-xl text-gray-900 dark:text-dark-txt placeholder:text-gray-800 focus:placeholder:text-gray-400 dark:bg-dark-second dark:focus:text-gray-200 dark:placeholder:text-gray-400 dark:focus:placeholder:text-gray-500"
                 placeholder={`Whats on your mind, ${capitalize(firstName)}?`}
               />
+              {showDropzone && (
+                <ImageDropzone files={files} setFiles={setFiles}>
+                  <ImagePreview files={files} setFiles={setFiles} />
```

Comment:
This looks good man 👏 
Now I completely get what u explained previously about the children components. Great 🚀